### ISSUE TITLE
cldrjs is broken in the newest version

### ISSUE THREAD

**Issue by swernerx** [jspm/jspm-cli#1434]
I am not yet able to identify the affected version... but this is struggling me for a few hours now. Something strange what I identified.

[cldrjs](https://github.com/rxaviers/cldrjs) has the following code in `node_main.js` (main entry point for jspm) which contains this code:

```js
module.exports = require('./cldr');
require('./cldr/event');
require('./cldr/supplemental');
require('./cldr/unresolved');
```

Interestingly the three following scripts after `cldr` also try to require "cldrjs" (which should bave been created by the first "exports") for extending it.

event.js:

```js
"format cjs";
(function(factory) {
  if (typeof define === "function" && define.amd) {
    define(["../cldr"], factory);
  } else if (typeof module === "object" && typeof module.exports === "object") {
    module.exports = factory(require('../../../cldrjs@0.4.3'));
  } else {
    factory(Cldr);
  }
}(function(Cldr) {
```

This looks the same in all three files. Maybe the problem is some recursion prevention logic in JSPM... the part which breaks the code is that the require() to "cldrjs" in all these three files is being resolved into an empty object instead of the exported method (from cldr.js).

This is problematic because `event.js` is trying to access to `prototype` of the exported function:

```js
  superInit = Cldr.prototype.init;
  Cldr.prototype.init = function() {
    ...
  };
```

Is that anything someone else sees? Possibly in other libraries?

BTW: Somehow the same thing worked a few days ago.

**Comment by swernerx** [comment]
This is the error from the browser

![error-cldrjs](https://cloud.githubusercontent.com/assets/145201/12354715/fd6f9356-bb97-11e5-84cf-d72f72892db7.png)

**Comment by swernerx** [comment]
Whe I am changing the first few lines in `event.js` to use the "local"/"non-versioned" version of cldr.js instead it works:

```js
"format cjs";
(function(factory) {
  if (typeof define === "function" && define.amd) {
    define(["../cldr"], factory);
  } else if (typeof module === "object" && typeof module.exports === "object") {
    module.exports = factory(require('../cldr'));
  } else {
    factory(Cldr);
  }
}(function(Cldr) {
```

But then the next library in the inclusion order fails: jquery/globalize which seem to depend on the same pattern of extending the main module.

**Comment by swernerx** [comment]
```bash
$ jspm --version
0.16.23
Running against local jspm install.
```

**Comment by swernerx** [comment]
Somehow I hoped this will be fixed by going back to an older version e.g. via:

```bash
npm install jspm@0.16.15
jspm install -f
```

Unfortunately this does not change anything.

**Comment by swernerx** [comment]
The load order is actually looking good.

![load-order](https://cloud.githubusercontent.com/assets/145201/12355490/54cf35ac-bb9d-11e5-9b6f-42aaee6bc9d2.png)

**Comment by guybedford** [comment]
When you downgraded did you definitely run `jspm dl-loader --latest` to make sure it was using the right version of SystemJS? Would be good to know if this is a configuration issue or an actual regression.

**Comment by swernerx** [comment]
In this image (where I added some debug code to system.src.js one can see fine that there is a difference in the objects resolved from the two different requires. The first one returns the correct function while the second ones returns the problematic empty object.

![bildschirmfoto 2016-01-15 um 15 53 40](https://cloud.githubusercontent.com/assets/145201/12355957/53c1d96e-bba0-11e5-9084-c4dd919e6b31.png)

**Comment by swernerx** [comment]
@guybedford I have not tried that yet. Thanks!

**Comment by swernerx** [comment]
Okay, this also does not seem to help. :(

![going-back](https://cloud.githubusercontent.com/assets/145201/12356141/4483265a-bba1-11e5-8047-e553fbd0b606.png)

**Comment by swernerx** [comment]
I just tried to build a quick *new* app just using `cldrjs` but this also fails.

I uploaded the files to a Gist:
https://gist.github.com/swernerx/09760e33859aa0243f1c

**Comment by swernerx** [comment]
I have the same thing running on another computer. Interestingly the same code does work there fine.

```js
/* */ 
"format cjs";
(function(factory) {
  if (typeof define === "function" && define.amd) {
    define(["../cldr"], factory);
  } else if (typeof module === "object" && typeof module.exports === "object") {
    module.exports = factory(require('../node_main'));
  } else {
    factory(Cldr);
  }
}(function(Cldr) {
```

This is the same part from "cldrjs" like shown before, but the `require` line looks totally different. Requires the locale reference to `node_main` instead of the more project related `cldrjs@...` variant.

I am not sure why I was unable to produce the same result when installing the older version on my MacBook in the office.

The jspm version installed here is: 0.16.19

- nodejs: v5.4.0 (I figure in the office I am at v5.4.1)
- npm: v3.5.3
- SystemJS at: v0.19.9 (as found in jspm_packages)
- cldrjs: 0.4.3 (same)

Might this be somehow related to the NodeJS/npm version? The code is definitely being rewritten when installed by jspm. The question is who is responsible for the differences between both of these sections.

**Comment by swernerx** [comment]
I figured out something more.

- NodeJS seems not to be responsible (even switching back to older versions keeps the same phenomenon)

- It seems like the "-f" flag is somehow broken. Doing a `jspm cache-clear && rm -rf jspm_packages && jspm install` leads to a working version while `jspm install -f` produces the broken version.

This was tested on my home computer with jspm 0.16.19.

Does this makes any sense for you?

**Comment by swernerx** [comment]
BTW: This is definitely reproduceable. 

File event.js from cldrjs:

- With f-flag: `module.exports = factory(require('../../../cldrjs@0.4.3'));` (does not work)
- With cache-clean + install: `module.exports = factory(require('../node_main'));` (works fine)

**Comment by unional** [comment]
I suspect these two are related:
https://github.com/jspm/jspm-cli/issues/1448
Can you dig into event.js:240 to see if the affected code gets similar issue (the referenced module shape is different than in 0.16.20)?

**Comment by guybedford** [comment]
Fixed in https://github.com/jspm/jspm-cli/commit/028c753d4ebfe190858cfd7234037702cf954909.

**Comment by guybedford** [comment]
Fixed in jspm 0.16.25.
